### PR TITLE
Table Shuffle!

### PR DESCRIPTION
**Pull request by SuperSayu**
Adds the Table Shuffle subsystem from old Sayustation code, expanded and polished to be about 200% less terrible.

## About The Pull Request

The table shuffle subsystem runs at game init and moves things which are publicly accessible to similar locations within the mapping area they started in.  This can be modified in many ways, and it can be disabled with a flag in game_options.txt.  Other configuration options let you experiment with doing less or vastly more shuffling than default.

These, broadly, are the consequences:
* Vending machines may provide free items (less than a configurable cost).
* Items on tables or racks, or in unlocked cabinets and crates, may be moved.
* Items in a storage item (boxes, in particular) may be moved.
* Food and drink that finds itself an item of interest (vended or moving) may be consumed partly or fully.
* Moving items may end up on top of a table, rack, chair, sink, toilet, shower, scrubber, or on top of a number of other conspicuous machines, where they should still be visible.  They may also be placed in unlocked closets.
* Items on tables and racks may be moved slightly (pixel offset changed, stack order on the table changed).

Which of these things happens within a given area can be modified through a var on that area, so that surgery tools are not moved out of place and the nuke disk is never moved to the completely walled-in rack next to the Captain's quarters.

To assist administrators with changing and observing the consequences of these settings, debug verbs and logging have been added:
* You can observe a detailed log of everything that happened due to table shuffle code
* You can view a list of "high roller" areas that had more than a set number of changes occur, with links in that list to view the logs of the affected areas
* You can manually shuffle an area, with the area shuffle log drawing a line between shuffles to avoid confusion
* You can do a "full" shuffle of an area that ignores the area shuffle options, and observe potential consequences.

In testing, table shuffle code has never taken more than a half-second during initialization (but I have not done a stress-test with insanely high probabilities).

The only files that can conflict with the upstream code (unless I pass this system to them) are the .dme, admin_verbs.dm, and game_options.txt .  Everything else is in an orbstation-specific file.

## Why It's Good For The Game

I originally developed table shuffle shortly after I developed dirty floor code.  It's my assertion that everything being the same every round is boring.  These changes are minimal, but more importantly, they MOSTLY appear organic; something is picked up and put down elsewhere, something is eaten or half-drunk.  Vending machine items are properly vended, meaning that the vender has less at roundstart because of the janitor or whoever that had it before the game started.

Meanwhile, you may notice things that you pass over because they're usually irrelevant to you, especially when a vending machine spits out clothing or items that you never care about, and it's suddenly sitting in the open where you can see it.  Things that are usually in boxes are scattered out, drawing your eye.  Piles of stuff you know is meaningful to another job but may not be useful for yours will be in the open for you to trip over (not literally).

In short, it creates some of the chaos that players normally cause, without players actually being involved, and without serious consequences.

## Changelog

:cl:
add: Added table shuffle system
config: Table shuffle configuration options added to game_options.txt
admin: Several debug verbs added to audit and trigger the table shuffle system.
/:cl:

### CODE REVIEW

**Comment by SuperSayu** [comment]
These are what the logs look like (trimmed)
image

**Review by Jacquerel — APPROVED** [review]
Seems pretty good to me.

My only comments are style based. It would be nice to have more autodoc style comments for procs and variable definitions so that they reliably show up in vscode intellisense.
`/// comment` or `/** */` comment.

It would also be preferable to remove single letter or two letter variable names in place of descriptive ones, some of them get a little hard to follow and while we don't actually need to follow tg's style guide I think that's a good rule.

**Comment by SuperSayu** [comment]
I'll keep those in mind if I end up pushing this upstream
